### ISSUE TITLE
Matplotlib update to version 2.0

### ISSUE THREAD

**Issue by khs26** [pele-python/pele#139]
A placeholder issue for `matplotlib` changes that might break our existing calls. Not sure exactly which changes might need making and I don't currently have time to test it out.

**Comment by smcantab** [comment]
my opinion is that matplotlib should typically be wrapped in a `try`
statement, since one typically does not uses matplotlib unless he is
 analysing the data. That way it won't break anything if the import is not
resolved at run time, and one can fix it at a later stage during data
analysis if need to

**Comment by khs26** [comment]
I agree with that. Especially given that it can take some fiddling with the various graphical backends to get it to work reliably. Does the GUI use `matplotlib` for things that we need to work reliably?

**Comment by smcantab** [comment]
yes the gui relies heavily on matplotlib, and as far as I can tell it uses
exclusively the QT4Agg backend. As far as I can tell the whole point of the
gui is that one wants to visualise results, so I don't see how you can
bypass matplotlib when using the gui. What I mean is that if there is a
matplotlib import in the NEB source because there is a test snippet at the
end of the script then that should be wrapped in a try statement because it
is not strictly necessary to run a NEB calculation.

**Comment by smcantab** [comment]
re-reading my post I noticed that it might sound a bit passive-aggressive,
it wasn't my intention though! (I was just writing in a bit of a rush) let
me know if I can help on a particular issue related to this
